packages/mobile/src/store/products/selectors.ts

```diff
@@ -0,0 +1,11 @@
+import { RootState } from '~/common/types/types';
+import { createSelector } from '@reduxjs/toolkit';
+
+const selectProductById = createSelector(
+  (state: RootState) => state.products.products,
+  (state: RootState, productId: string) => productId,
+
+  (products, productId) => products.find((item) => item.id === productId),
+);
```

Comment:
```suggestion
  [
      (state: RootState) => state.products.products,
      (state: RootState, productId: string) => productId
  ],
  (products, productId) => products.find((item) => item.id === productId),
);
```

Also this part - `(state: RootState) => state.products.products,` - could be extracted to a separate selector, e.g. `selectProducts`

Reply:
solved